extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -427,24 +430,63 @@ export abstract class AzureAuth implements vscode.Disposable {
 		}
 	}
 
-	private async openConsentDialog(tenantId: string, resourceId: string): Promise<boolean> {
+	private async openConsentDialog(tenant: Tenant, resourceId: string): Promise<boolean> {
+		if (!tenant.displayName && !tenant.id) {
+			throw new Error('Tenant did not have display name or id');
+		}
+
+		if (tenant.id === 'common') {
+			throw new Error('Common tenant should not need consnet');
+		}
+
+		const getTenantConfigurationSet = (): Set<string> => {
+			const configuration = vscode.workspace.getConfiguration('azure.tenant.config');
+			let values: string[] = configuration.get('filter') ?? [];
+			return new Set<string>(values);
+		};
+
+		const updateTenantConfigurationSet = (set: Set<string>): void => {
```

Comment:
This can be moved below the if statement right below since we'll just be returning immediately if it's true